tools/migration/internal/rmh_whitebox_test.go

```diff
@@ -0,0 +1,46 @@
+package internal
+
+import (
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+
+	ast "github.com/stretchr/testify/assert"
+	req "github.com/stretchr/testify/require"
+
+	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
+)
+
+func TestGetNewRepoPath(t *testing.T) {
+	tf.UnitTest(t)
+	assert := ast.New(t)
+	require := req.New(t)
+
+	t.Run("Uses the new repo opt as a prefix if provided", func(t *testing.T) {
+		dirname := "/tmp/mynew_repodir"
+		newpath := getNewRepoPath("/tmp/myold_repodir", dirname, "1", "2")
+		rgx, err := regexp.Compile("/tmp/mynew_repodir_1_2_[0-9]{4}-[0-9]{2}-[0-9]{2}_[0-9]{6}")
+		require.NoError(err)
+		assert.Regexp(rgx, newpath)
+	})
+
+	t.Run("Adds a timestamp to the new repo dir", func(t *testing.T) {
+		dirname := "/tmp/myfilecoindir"
+		newpath := getNewRepoPath(dirname, "", "1", "2")
+		rgx, err := regexp.Compile("/tmp/myfilecoindir_1_2_[0-9]{4}-[0-9]{2}-[0-9]{2}_[0-9]{6}")
+		require.NoError(err)
+		assert.Regexp(rgx, newpath)
+	})
```

Comment:
Separating these "white box" tests from the interface tests makes things a bit harder to follow, and I don't think is achieving much. They're exercising functionality that is declared by the interfaces/comments: i.e. externally observable. The directory names are observable from `File.Name()`.

Roll these tests in with `repo_migration_helper_test.go` so one place provides a comprehensive exercise of the functionality.

Reply:
I initially wrote unit tests trying to observe from `File.Name()` and one of the problems this causes, is you must spend a lot of time ensuring that files are created, ensuring that they are deleted afterward. Fully testing this function through an exported function is what drove the question about testing with environment variables, when what I really wanted to test was the behavior of getNewRepoPath, not the ability to create and destroy files.